src/ops/array.rs

```diff
@@ -15,42 +15,46 @@ pub(crate) fn rescale_internal(value: &mut [u32; 3], value_scale: &mut u32, new_
     }
 
     if *value_scale > new_scale {
-        let mut diff = *value_scale - new_scale;
+        let mut diff = value_scale.wrapping_sub(new_scale);
         // Scaling further isn't possible since we got an overflow
         // In this case we need to reduce the accuracy of the "side to keep"
 
         // Now do the necessary rounding
         let mut remainder = 0;
-        while diff > 0 {
+        while let Some(diff_minus_one) = diff.checked_sub(1) {
```

Comment:
The same branch condition of `diff > 0`. Removes one arithmetic operation and assembly output shouldn't change